firmware/temperature_detection/temperature_detection.ino

```diff
@@ -58,31 +60,40 @@ void loop() {
         display_status(STATUS_DISTANCE_RIGHT);
         delay(time_before_sensor_stab); // Wait for the sensor to stabilize.
         Serial.println("Temperature measurement");
-        Serial.println(measure_temperature());
+        Serial.println(measure_temperature(temperature_samples));
         display_status(
-            (measure_temperature() > limit_fever)
+            (measure_temperature(temperature_samples) > limit_fever)
             ? STATUS_FEVER_HIGH : STATUS_FEVER_LOW
         );
         // Wait for the person to go away...
-        while(measure_distance() <= next_person_distance) delay(time_before_leaving);
+        while(measure_distance(distance_samples) <= next_person_distance) {
+            delay(time_before_leaving);
+        }
     }
 }
 
 
-int measure_distance() {
-    digitalWrite(ultrasound_trigger_pin, LOW);
-    delayMicroseconds(2);
-    digitalWrite(ultrasound_trigger_pin, HIGH);
-    delayMicroseconds(10);
-    digitalWrite(ultrasound_trigger_pin, LOW);
-    long duration = pulseIn(ultrasound_echo_pin, HIGH);
-    int distance = duration * 0.034 / 2;
-    return distance; // in centimeters
+int measure_distance(size_t samples) {
+    float measurements = 0;
+    for(size_t counter = 0; counter < (samples || 1); counter++) {
+        digitalWrite(ultrasound_trigger_pin, LOW);
+        delayMicroseconds(2);
+        digitalWrite(ultrasound_trigger_pin, HIGH);
+        delayMicroseconds(10);
+        digitalWrite(ultrasound_trigger_pin, LOW);
+        long duration = pulseIn(ultrasound_echo_pin, HIGH);
+        measurements += duration * 0.034 / 2;
+    }
+    return measurements / (samples || 1); // average in centimeters
 }
 
 
-float measure_temperature() {
-    return temperature_sensor.readObjectTempC(); // in celsius degrees
+float measure_temperature(size_t samples) {
+    float measurements = 0;
+    for(size_t counter = 0; counter < (samples || 1); counter++) {
+        measurements += temperature_sensor.readObjectTempC();
+    }
```

Comment:
I think we should add a Settling time for the sensor between each readings, as a first approach it could be something like 250ms

Reply:
There is more information here:
https://www.melexis.com/-/media/files/documents/datasheets/thermometer-datasheet-melexis.pdf
8.4.7. Timing specification

Reply:
```suggestion
    for(size_t counter = 0; counter < (samples || 1); counter++) {
        measurements += temperature_sensor.readObjectTempC();
        delay(interval);
    }
```

Maybe we can just update the function signature and add an `interval` parameter.

Reply:
Yes it would be perfect